autograd/numpy/fft.py

```diff
@@ -10,18 +10,82 @@
 
 # TODO: make fft gradient work for a repeated axis,
 # e.g. by replacing fftn with repeated calls to 1d fft along each axis
-def fft_defvjp(fft_fun):
-    def fft_grad(g, ans, vs, gvs, x, *args, **kwargs):
-        check_no_repeated_axes(*args, **kwargs)
-        return match_complex(vs, truncate_pad(fft_fun(g, *args, **kwargs), vs.shape))
-    fft_fun.defvjp(fft_grad)
+def fft_grad(get_args, fft_fun, g, ans, vs, gvs, x, *args, **kwargs):
+    axes, s, norm = get_args(x, *args, **kwargs)
+    check_no_repeated_axes(axes)
+    return match_complex(vs, truncate_pad(fft_fun(g, *args, **kwargs), vs.shape))
 
-for fft_fun in (fft, ifft, fft2, ifft2, fftn, ifftn):
-    fft_defvjp(fft_fun)
+fft.defvjp(lambda *args, **kwargs:
+        fft_grad(get_fft_args, fft, *args, **kwargs))
+ifft.defvjp(lambda *args, **kwargs:
+        fft_grad(get_fft_args, ifft, *args, **kwargs))
+
+fft2.defvjp(lambda *args, **kwargs:
+        fft_grad(get_fft_args, fft2, *args, **kwargs))
+ifft2.defvjp(lambda *args, **kwargs:
+        fft_grad(get_fft_args, ifft2, *args, **kwargs))
+
+fftn.defvjp(lambda *args, **kwargs:
+        fft_grad(get_fft_args, fftn, *args, **kwargs))
+ifftn.defvjp(lambda *args, **kwargs:
+        fft_grad(get_fft_args, ifftn, *args, **kwargs))
+
+def rfft_grad(get_args, irfft_fun, g, ans, vs, gvs, x, *args, **kwargs):
+    axes, s, norm = get_args(x, *args, **kwargs)
+
+    check_no_repeated_axes(axes)
+    if s is None: s = [vs.shape[i] for i in axes]
+    check_even_shape(s)
+
+    # s is the full fft shape
+    # gs is the compressed shape
+    gs = list(s)
+    gs[-1] = gs[-1] // 2  + 1
+    fac = make_rfft_factors(axes, gvs.shape, gs, s, norm)
+    g = anp.conj(g / fac)
+    r = match_complex(vs, truncate_pad((irfft_fun(g, *args, **kwargs)), vs.shape))
+    return r
+
+    rfft_fun.defvjp(rfft_grad)
```

Comment:
The line is not supposed to be there. It has no effect and must have slipped through editing.